packages/manager/apps/pci-kubernetes/src/api/hooks/nodes.ts

```diff
@@ -0,0 +1,168 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { PaginationState } from '@ovhcloud/manager-components';
+import { applyFilters, Filter } from '@ovh-ux/manager-core-api';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useKubernetesCluster } from '@/api/hooks/useKubernetes';
+import { useRegionFlavors } from '@/api/hooks/flavors';
+import { deleteNode, getNodes, TNode } from '@/api/data/nodes';
+import { useInstances } from '@/api/hooks/instances';
+import { switchToMonthlyBilling } from '@/api/data/instance';
+
+export const useNodes = (
+  projectId: string,
+  clusterId: string,
+  nodePoolId: string,
+) =>
+  useQuery({
+    queryKey: [
+      'project',
+      projectId,
+      'kubernetes',
+      clusterId,
+      'nodePools',
+      nodePoolId,
+      'nodes',
+    ],
+    queryFn: () => getNodes(projectId, clusterId, nodePoolId),
+    throwOnError: true,
+  });
+
+const paginateResults = <T>(items: T[], pagination: PaginationState) => ({
+  rows: items.slice(
+    pagination.pageIndex * pagination.pageSize,
+    (pagination.pageIndex + 1) * pagination.pageSize,
+  ),
+  pageCount: Math.ceil(items.length / pagination.pageSize),
+  totalRows: items.length,
+});
+
+export const defaultCompareFunction = (key: keyof TNode) => (
+  a: TNode,
+  b: TNode,
+) => {
+  const aValue = a[key] || '';
+  const bValue = b[key] || '';
+
+  return aValue.toString().localeCompare(bValue.toString());
+};
```

Comment:
Maybe make it generic while it's also used on `pci-kubernetes/src/api/hooks/node-pools.ts`